flink/src/main/java/org/apache/iceberg/flink/source/RowDataIterator.java

```diff
@@ -92,12 +102,12 @@
     return iter;
   }
 
-  private CloseableIterable<RowData> newAvroIterable(FileScanTask task, Map<Integer, ?> idToConstant) {
+  private CloseableIterable<RowData> newAvroIterable(FileScanTask task, Schema schema, Map<Integer, ?> idToConstant) {
     Avro.ReadBuilder builder = Avro.read(getInputFile(task))
-        .reuseContainers()
-        .project(projectedSchema)
+        .reuseContainers(false)
```

Comment:
@JingsongLi , This is used to fix the UT.  How can we copy the `RowData`?

Reply:
@chenjunjiedada can you please clarify for me what `the UT` stands for?

Reply:
I mean the unit tests in `TestFlinkScan`.  The `getRows` puts each `row` from `inputformat.nexRecord(null)` into List while the `row` is reused when the file format is Avro, so the result in List is wrong.

I didn't find a simple way to copy the `GenericRowData`, so I set `reuseContainer` to false to align with Parquet and ORC cases.  I changed it back in 8526b6deedb747200fb21507f5716cf5e8396470 since I found the converter could be used to copy the `row`. But there comes new concern about the double copies for Parquet and ORC. @openinx @JingsongLi @rdblue ,  Should we reuse the container for Flink read?

Reply:
The `inputformat.nexRecord` returns reused record, it is OK in Flink.

Reply:
I think we can create a PR to reuse Parquet container for Flink and Spark.

Reply:
@JingsongLi , Thanks for your comments. I think it would be better to use an option to set `reuse`, let me create one.

Reply:
I have a slight -1 for option, If there are no side effects, why do we need to provide this option? (testing is not a good example, we should consider user-face interface)

Reply:
And if this reuse flap is false, I think there may also be some risks.
Note in Flink and Spark reader, we are reusing binary for StringReader. 
Maybe these binaries are chunk buffers that have been reused by parquet reader (CC: @rdblue), so even if reuse flag is false, users cannot assume returning row's security.

Reply:
@JingsongLi, I created https://github.com/apache/iceberg/pull/1522. We could also discuss there.